src/main/java/io/hops/hopsworks/expat/migrations/elk/BaseIndexTemplateMigrate.java

```diff
@@ -0,0 +1,158 @@
+      + "\"public_ds\":{\"type\":\"boolean\" },"
+      + "\"description\":{\"type\":\"text\" },"
+      + "\"name\":{\"type\":\"text\" },"
+      + "\"parent_id\":{\"type\":\"long\" },"
+      + "\"partition_id\":{ \"type\":\"long\" },"
+      + "\"user\":{ \"type\":\"keyword\" },"
+      + "\"group\":{ \"type\":\"keyword\" },"
+      + "\"operation\":{ \"type\":\"short\" },"
+      + "\"size\":{ \"type\":\"long\" },"
+      + "\"timestamp\":{ \"type\":\"long\" },"
+      + "\"xattr\":{ \"type\":\"nested\", \"dynamic\":true }"
+      + "}}";
+  
+  private static final String FEATURESTORE_INDEX = "featurestore";
+  private static final String FEATURESTORE_MAPPING =
+    "\"mappings\":{"
+      + "\"dynamic\":\"strict\","
+      + "\"properties\":{"
+      + "\"doc_type\":{\"type\":\"keyword\"},"
+      + "\"name\":{\"type\":\"text\"},"
+      + "\"version\":{\"type\":\"integer\"},"
+      + "\"project_id\":{\"type\":\"integer\"},"
+      + "\"project_name\":{\"type\":\"text\"},"
+      + "\"dataset_iid\":{\"type\":\"long\"},"
+      + "\"xattr\":{\"type\":\"nested\",\"dynamic\":true}"
+      + "}}";
+  
+  private static final String APP_PROVENANCE_INDEX = "app_provenance";
+  private static final String APP_PROVENANCE_MAPPING =
+    "\"mappings\":{"
+      + "\"properties\":{"
+      + "\"app_id\":{\"type\":\"keyword\"},"
+      + "\"app_state\":{\"type\":\"keyword\"},"
+      + "\"timestamp\":{\"type\":\"long\"},"
+      + "\"app_name\":{\"type\":\"text\"},"
+      + "\"app_user\":{\"type\":\"text\"}"
+      + "}}";
+
+  private void setup()
+    throws ConfigurationException, KeyStoreException, NoSuchAlgorithmException, KeyManagementException {
+    
+    Configuration conf = ConfigurationBuilder.getConfiguration();
+    
+    String elasticURI = conf.getString(ExpatConf.ELASTIC_URI);
+    if (elasticURI == null) {
+      throw new ConfigurationException(ExpatConf.ELASTIC_URI + " cannot be null");
+    }
+    elastic = HttpHost.create(elasticURI);
+    
+    elasticUser = conf.getString(ExpatConf.ELASTIC_USER_KEY);
+    if (elasticUser == null) {
+      throw new ConfigurationException(ExpatConf.ELASTIC_USER_KEY + " cannot be null");
+    }
+    elasticPass = conf.getString(ExpatConf.ELASTIC_PASS_KEY);
+    if (elasticPass == null) {
+      throw new ConfigurationException(ExpatConf.ELASTIC_PASS_KEY + " cannot be null");
+    }
+    
+    httpClient = HttpClients
+      .custom()
+      .setDefaultRequestConfig(RequestConfig.custom().setCookieSpec(
+        CookieSpecs.IGNORE_COOKIES).build())
+      .setSSLContext(new SSLContextBuilder().loadTrustMaterial((x509Certificates, s) -> true).build())
+      .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+      .build();
+  }
+  
+  @Override
+  public void migrate() throws MigrationException {
+    try {
+      LOGGER.info("setup");
+      setup();
+      LOGGER.info("templates");
+      ElasticClient.createTemplate(httpClient, elastic, elasticUser, elasticPass,
```

Comment:
does it affect existing indices if you create the template after?

Reply:
No, templates do not affect created indices. It only affects when we drop the index and have epipe reindex.